packages/flutter/lib/src/widgets/scroll_absolute.dart

```diff
@@ -94,9 +94,16 @@ class ViewportScrollBehavior extends ScrollBehavior2 {
     return null;
   }
 
+  ScrollPhysics _getEffectiveScrollPhysics(BuildContext context, ScrollPhysics physics) {
+    final ScrollPhysics defaultPhysics = getScrollPhysics(getPlatform(context));
+    if (physics is ScrollPhysicsProxy)
+      return physics.applyTo(defaultPhysics);
+    return defaultPhysics;
+  }
```

Comment:
I think my biggest problem here is with the "physics is ScrollPhysicsProxy" check. I'd feel less bad about this if ScrollPhysics objects in general had a concept of chaining, and you just always called `physics.chain(defaultPhysics)` or some such.